book/styles.md

```diff
@@ -560,115 +705,137 @@ def style(node, rules):
         # ...
 ```
 
-Now that we have `font-weight` and `font-style` set on every node, we
-can use them in `InlineLayout` to set the font:
+With `font-weight` and `font-style` set on every node, `InlineLayout`
+no longer needs `style`, `weight`, and `size` fields; they were only
+there to track when text was inside or outside `<i>` and `<b>` tags,
+and now styles and inheritance are doing that job:
 
 ``` {.python}
-self.bold = node.style.get("font-weight", "normal") == "bold"
-self.italic = node.style.get("font-style", "normal") == "italic"
+class InlineLayout:
+    def font(self):
+        bold = node.style["font-weight"]
+        italic = node.style["font-style"]
+        if italic == "normal": italic = "roman"
+        size = int(px(node.style.get("font-size")) * .75)
+        return tkinter.font.Font(size=size, weight=weight, slant=slant)
+    
+    def text(self, text):
+        font = self.font()
+        for word in text.split():
+            # ...
 ```
 
-Now that we have styles on both block and inline nodes, we can
-refactor `is_inline`. Instead of testing directly for the `<b>` and
-`<i>`, we could test for a CSS property both share. The standard
-property is `display`, which can be either `block` or `inline`; it
-basically tells you which of the two layout modes to use.^[Modern CSS
-adds some funny new values, like `run-in` or `inline-block`, and it
-has layout modes set by other properties, like `float` and
-`position`. Nothing gold can stay.] So instead of...
+Note that the `font-style` needs to replace the CSS default of
+"normal" with the Tk value "roman", and the `font-size` needs to be
+converted from points to pixels.[^72ppi]
 
-``` {.python}
-return node.tag in ["i", "b"]
-```
+[^72ppi]: Normally you think of points as a physical length unit (one
+    72^nd^ of an inch) and pixels as a digital unit (dependent on the
+    screen) but in CSS, the conversion is fixed at exactly 75% (or 96
+    pixels per inch). I'm not sure why, it seems weird and it does
```

Comment:
The reason is to ensure existing sites are readable on all devices.

Reply:
Historically, it seems to have gone like this:

1. most/all monitors had similar pixel densities
2. sites were build-and-forget
3. CSS introduced the 'zoom' property, which allowed scaling outside of the monitor, and laid bare the non-physicality of "physical" units
4. High-DPI/"retina" displays futher moved way from true "phyisical"